OpenRA.Game/UtilityCommands/ExtractChromeStrings.cs

```diff
@@ -0,0 +1,124 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+
+namespace OpenRA.UtilityCommands
+{
+	sealed class ExtractChromeStringsCommand : IUtilityCommand
+	{
+		string IUtilityCommand.Name { get { return "--extract-chrome-strings"; } }
+
+		bool IUtilityCommand.ValidateArguments(string[] args)
+		{
+			return true;
+		}
+
+		[Desc("Extract translatable strings that are not yet localized and update chrome layout.")]
+		void IUtilityCommand.Run(Utility utility, string[] args)
+		{
+			// HACK: The engine code assumes that Game.modData is set.
+			var modData = Game.ModData = utility.ModData;
+
+			var types = modData.ObjectCreator.GetTypes();
+			var translatableFields = types.SelectMany(t => t.GetFields())
+				.Where(f => f.HasAttribute<TranslationReferenceAttribute>()).Distinct();
+
+			foreach (var chromeLayout in modData.Manifest.ChromeLayout)
+			{
+				modData.ModFiles.TryGetPackageContaining(chromeLayout, out var chromePackage, out var chromeName);
+				var chromePath = chromePackage.Name + Path.DirectorySeparatorChar + chromeName;
+
+				var fluent = chromeLayout.Split('|').First() + "|languages/chrome/en.ftl";
```

Comment:
`string.Split` should return an array, so
```suggestion
				var fluent = chromeLayout.Split('|')[0] + "|languages/chrome/en.ftl";
```

Also, this is really weird as I assume that file does not exist yet. Is there no other way to get the fluent package? E.g. directly calling `OpenPackage`?